compose/service/module.go

```diff
@@ -443,19 +444,46 @@ func (svc module) FindSensitive(ctx context.Context, filter types.PrivacyModuleF
 
 		for _, m := range mm {
 			isPrivate := false
+			// Check module field's sensitivity level
 			for _, f := range m.Fields {
 				if !isPrivate {
-					isPrivate = f.Private
+					isPrivate = f.Privacy.SensitivityLevel > 0
 				}
 			}
 
+			// Check module's sensitivity level
+			if !isPrivate {
+				isPrivate = m.Privacy.SensitivityLevel > 0
+			}
+
+			// Check sensitivity level of connection
+			cMeta, err := svc.dal.GetConnectionMeta(ctx, m.ModelConfig.ConnectionID)
+			if err != nil {
+				return err
+			}
+
+			if !isPrivate {
+				isPrivate = cMeta.SensitivityLevel > 0
+			}
+
 			if isPrivate && m != nil {
-				set = append(set, types.PrivacyModule{
+				pm := types.PrivacyModule{
 					ID:           m.ID,
 					Name:         m.Name, // @todo get this as per translation
 					Handle:       m.Handle,
+					NamespaceID:  m.NamespaceID,
 					ConnectionID: m.ModelConfig.ConnectionID,
-				})
+				}
+
+				// Set default connection ID
+				if pm.ConnectionID == 0 {
+					pm.ConnectionID, err = svc.dal.GetDefaultConnectionID(ctx)
```

Comment:
Here, I would add a new `ConnectionID` field to the `ConnectionMeta` struct and use `cMeta`